.github/workflows/docs_changes_summary.yml

```diff
@@ -67,3 +73,42 @@ jobs:
 
             ${{steps.get-changes.outputs.result}}
           edit-mode: replace
+      - name: Mark as breaking
+        id: breaking
+        if: contains(steps.get-changes.outputs.result, "BREAKING CHANGE")
+        run: echo "status=true" >> $GITHUB_OUTPUT
+  ensure-breaking-changes-released-as-major-bump:
+    runs-on: ubuntu-latest
+    needs: [doc-changes]
+    steps:
+      - name: Should enforce check
+        id: enforce
+        # Only enforce this check for renovate PRs that update dependencies
+        if: needs.doc-changes.outputs.breaking == 'true' && github.event.pull_request.user.login == 'cq-bot' && startsWith(github.event.pull_request.title, 'fix(deps)') && startsWith(github.head_ref, 'renovate/')
+        run: echo "status=true" >> $GITHUB_OUTPUT
+      - name: Install commit message parser
+        if: steps.enforce.outputs.status == 'true'
+        run: npm install @conventional-commits/parser
+      - uses: actions/github-script@v6
+        if: steps.enforce.outputs.status == 'true'
+        with:
+          script: |
+            const {
+              parser,
+              toConventionalChangelogFormat,
+            } = require("@conventional-commits/parser");
```

Comment:
I chose this package as this is what `release-please` uses https://github.com/googleapis/release-please/blob/f10f3b49a0cd048f84e2fcd399bbcee74bd3f47c/package.json#L70